### PR TITLE
test site replaced by web app test site

### PR DESCRIPTION
**Pull request by frenzyigor**
Test Web Site converted to Web Application

### CODE REVIEW

File: sources/LibProtection.TestSite/Default.aspx.cs [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+            return builder.ToString();
+        }
+
+        protected static string XMLTagBuilder(string xml)
+        {
+            if (string.IsNullOrWhiteSpace(xml)) { return string.Empty; }
+
+            var settings = new System.Xml.XmlReaderSettings();
+            settings.DtdProcessing = System.Xml.DtdProcessing.Parse;
+            var xmlDoc = new System.Xml.XmlDocument();
```
Comment (kochetkov) [inline sources/LibProtection.TestSite/Default.aspx.cs]: In .NET 4.6+ xmlDoc.XmlResolver is set to null by default. To make XXE possbile here you should set xmlDoc.XmlResolver to new XmlUrlResolver (https://msdn.microsoft.com/ru-ru/library/system.xml.xmlurlresolver(v=vs.110).aspx) instance.

File: sources/LibProtection.TestSite/Default.aspx.cs [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+            var xmlBytes = Encoding.ASCII.GetBytes(xml);
+
+            var doc = new System.Xml.XmlDocument();
```
Comment (kochetkov) [inline sources/LibProtection.TestSite/Default.aspx.cs]: Duplicate instantiation (already instantiated in xmlDoc before).

File: sources/LibProtection.TestSite/Default.aspx.cs [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+                        adapter.SelectCommand = command;
+
+                        using (var dataSet = new System.Data.DataSet())
+                        {
+                            adapter.Fill(dataSet);
+
+                            var builder = new System.Text.StringBuilder();
+                            foreach (System.Data.DataRow row in dataSet.Tables[0].Rows)
+                            {
+
```
Comment (kochetkov) [inline sources/LibProtection.TestSite/Default.aspx.cs]: Redundant line feed.

**Review by kochetkov — CHANGES_REQUESTED** [review]

File: sources/LibProtection.TestSite/Default.aspx.cs (hunk 1 above)
Comment (frenzyigor) [reply sources/LibProtection.TestSite/Default.aspx.cs]: fixed

File: sources/LibProtection.TestSite/Default.aspx.cs (hunk 2 above)
Comment (frenzyigor) [reply sources/LibProtection.TestSite/Default.aspx.cs]: fixed

File: sources/LibProtection.TestSite/Default.aspx.cs (hunk 3 above)
Comment (frenzyigor) [reply sources/LibProtection.TestSite/Default.aspx.cs]: fixed

**Review by kochetkov — APPROVED** [review]

**Review by kochetkov — APPROVED** [review]

**Pull request merged**